docs/design/ovs-pipeline.md

```diff
@@ -183,1015 +269,1363 @@ spec:
+```
+
+This ACNP is applied to all Pods with the `app: nginx` label in all Namespaces. For these Pods, it only allows TCP traffic
+on port 80 from all Pods with the label `app: nginx` and drop others. In addition to the ingress rules, this policy only
+allows egress TCP traffic on port 80 to all Pods with the label `app: nginx` and reject others. Similar to Kubernetes
+NetworkPolicy, Antrea will only install OVS flows for this policy on Nodes for which some of the Pods are the target of
+the policy. We still use `app:nginx` Pods mentioned above as the target of this policy.
+
+## Antrea L7 NetworkPolicy Implementation
+
+In addition to layer 3 and layer 4 policies mentioned above, [Antrea L7 NetworkPolicy](../antrea-l7-network-policy.md)
+is also supported in Antrea. The only difference is that Antrea L7 NetworkPolicy uses L7 protocol to match traffic, not
+layer 3 and layer 4 protocol.
+
+Consider the following Antrea L7 NetworkPolicy (AL7NP) in the Application tier as an example for the remainder of this
+document.
+
+```yaml
+apiVersion: crd.antrea.io/v1beta1
+kind: ClusterNetworkPolicy
+metadata:
+  name: cnp0
+spec:
+  priority: 10
+  tier: application # defaults to application tier if not specified
+  appliedTo:
+    - podSelector:
+        matchLabels:
+          app: nginx
+  ingress:
+    - action: Allow
+      from:
+        - podSelector:
+            matchLabels:
+              app: nginx
+      l7Protocols:
+        - http:
+            path: "/api/v2/*"
+            host: "foo.bar.com"
+            method: "GET"
+    - action: Drop
+  egress:
+    - action: Allow
+      to:
+        - podSelector:
+            matchLabels:
+              app: nginx
+      l7Protocols:
+        - http:
+            path: "/api/v2/*"
+            host: "foo.bar.com"
+            method: "GET"
+    - action: Reject
 ```
 
-This ACNP is applied to all Pods with the `app: server` label in all
-Namespaces. For these Pods, it drops TCP traffic on port 80 from all
-Pods which have the `app: notClient` label. In addition to the ingress rules,
-this policy also allows egress UDP traffic on port 53 to all Pods with the
-label `app: dns`. Similar to K8s NetworkPolicy, Antrea will only install OVS
-flows for this ACNP on Nodes for which some of the Pods are the target of the
-policy. Thus, we have scheduled three Pods (appServer, appDns, appNotClient)
-on the same Node and they have the following IP addresses:
+[TrafficControl](../traffic-control.md) is a CRD API that manages and manipulates the transmission of Pod traffic.
+Antrea creates additional dedicated table [TrafficControl] implement TrafficControl. Use the provided TrafficControls as
+examples throughout the remainder of this document.
 
-- appServer: 10.10.1.6
-- appNotClient: 10.10.1.7
-- appDns: 10.10.1.8
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: TrafficControl
+metadata:
+  name: tc-redirect
+spec:
+  appliedTo:
+    podSelector:
+      matchLabels:
+        nginx: redirect
```

Comment:
I don't think `nginx` is a meaningful label name
Even though this is an example, with a sole purpose of matching the CR to the corresponding flows, meaningful examples help the reader understand what's going on more easily